### PR TITLE
Cypress v10

### PR DESCRIPTION
**Pull request by josechirivella**
## Solution
- Upgraded `@vue/cli` tools to v5 from v3-v4
- Upgraded Cypress to v10 (latest is v12)
- Renamed the e2e files to `cy.js` since Cypress wanted that
- Cypress Dashboard already integrated
- Reduced the pipeline timing for about 2min compared to the current develop
- Upgraded eslint package
- Removed old packages that are not needed anymore

## How to Test
Test the steps above

## Related Tickets & Packages
- 

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

### CODE REVIEW

**Review by danloa — APPROVED** [review]

**Review by devmiguelangel — APPROVED** [review]

File: .browserlistrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+> 1%
```
Comment (rodriquelca) [inline .browserlistrc:1]: Please review the supported web browsers scope.
https://docs.processmaker.com/using/web-browser-requirements
https://browsersl.ist/#q=%3E++1%25%0Alast+2+versions%0Anot+dead

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,115 +1,246 @@
...
+    - cypress_tests_linux:
+        name: "UI Tests - Chrome"
+        browser: chrome
+        specPattern: "tests/e2e/**/*"
+        ciBuildId: ${CIRCLE_SHA1:0:8}
+        group: "UI - Chrome"
+        requires:
+          - Setup Linux
+    # Run E2E tests in Firefox
+#    - cypress_tests_linux:
```
Comment (rodriquelca) [inline .circleci/config.yml:236]: remove commented code

File: checkForOnly.sh [hunk 3]
```diff
@@ -22,12 +22,12 @@ for file in $stagedFiles; do
     fi
   fi
 
-  numberOfOnlys=$(grep "\.only" $file | wc -l)
-  if [[ $numberOfOnlys -gt 0 ]]; then
-    echo "$file contains .only(s)"
-    grep -n "\.only" $file --color=auto
-    exit 255
-  fi
+#  numberOfOnlys=$(grep "\.only" $file | wc -l)
```
Comment (rodriquelca) [inline checkForOnly.sh:25]: remove commented code

File: cypress.config.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+const { defineConfig } = require('cypress');
+
+module.exports = defineConfig({
+  projectId: '3839s4',
+  video: false,
+  screenshotOnRunFailure: true,
+  fixturesFolder: 'tests/e2e/fixtures',
+  // integrationFolder: 'tests/e2e/specs',
```
Comment (rodriquelca) [inline cypress.config.js:8]: remove this line

File: cypress.config.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  screenshotsFolder: 'tests/e2e/screenshots',
+  videosFolder: 'tests/e2e/videos',
+  // supportFile: 'tests/e2e/support/index.js',
```
Comment (rodriquelca) [inline cypress.config.js:11]: remove this line

File: tests/e2e/plugins/index.js [hunk 6]
```diff
@@ -3,14 +3,8 @@ module.exports = (on, config) => {
     config.baseUrl = 'https://processmaker.local.processmaker.com';
   }
 
-  on('task', require('@cypress/code-coverage/task'));
-  on('file:preprocessor', require('@cypress/code-coverage/use-browserify-istanbul'));
+  require('@cypress/code-coverage/task')(on, config);
+  // on('file:preprocessor', require('@cypress/code-coverage/use-browserify-istanbul'));
```
Comment (rodriquelca) [inline tests/e2e/plugins/index.js:7]: clean commented code

**Pull request merged**